exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -436,143 +542,336 @@ func (e *Exchange) GetWithdrawalsHistory(ctx context.Context, c currency.Code, _
 			})
-			if i == len(tradeData)-1 {
-				if ts.Equal(tt) {
-					// reached end of trades to crawl
-					break allTrades
-				}
-				if timestampStart.IsZero() {
-					break allTrades
-				}
-				ts = tt
-			}
 		}
 	}
-
-	err = e.AddTradesToBuffer(resp...)
-	if err != nil {
+	if err := e.AddTradesToBuffer(resp...); err != nil {
 		return nil, err
 	}
 	resp = trade.FilterTradesByTime(resp, timestampStart, timestampEnd)
-
 	sort.Sort(trade.ByDate(resp))
 	return resp, nil
 }
 
 // SubmitOrder submits a new order
 func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
-	if err := s.Validate(e.GetTradingRequirements()); err != nil {
+	if s == nil || *s == (order.Submit{}) {
+		return nil, common.ErrEmptyParams
+	}
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	oTypeString, err := OrderTypeString(s.Type)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
+	var tif string
+	tif, err = TimeInForceString(s.TimeInForce)
 	if err != nil {
 		return nil, err
 	}
-	return s.DeriveSubmitResponse(strconv.FormatInt(response.OrderNumber, 10))
+	switch s.AssetType {
+	case asset.Spot:
+		var smartOrder bool
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop:
+			smartOrder = true
+		case order.Limit, order.Market, order.LimitMaker, order.UnknownType:
+		default:
+			return nil, fmt.Errorf("%v order type %v is not supported", order.ErrTypeIsInvalid, s.Type)
+		}
+		if smartOrder {
+			var sOrder *PlaceOrderResponse
+			sOrder, err = e.CreateSmartOrder(ctx, &SmartOrderRequestRequest{
+				Symbol:        s.Pair,
+				Type:          oTypeString,
+				Side:          s.Side.String(),
+				AccountType:   accountTypeString(s.AssetType),
+				Price:         s.Price,
+				StopPrice:     s.TriggerPrice,
+				Quantity:      s.Amount,
+				ClientOrderID: s.ClientOrderID,
+				TimeInForce:   tif,
+			})
+			if err != nil {
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(sOrder.ID)
+		}
+		tif, err = TimeInForceString(s.TimeInForce)
+		if err != nil {
+			return nil, err
+		}
+		response, err := e.PlaceOrder(ctx, &PlaceOrderRequest{
+			Symbol:        s.Pair,
+			Price:         s.Price,
+			Amount:        s.Amount,
+			AllowBorrow:   false,
+			Type:          oTypeString,
+			Side:          s.Side.String(),
+			TimeInForce:   tif,
+			ClientOrderID: s.ClientOrderID,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return s.DeriveSubmitResponse(response.ID)
+	case asset.Futures:
+		var side string
+		var positionSide string
+		switch {
+		case s.Side.IsShort():
+			side = "SELL"
+			positionSide = "SHORT"
+		case s.Side.IsLong():
+			side = "BUY"
+			positionSide = "LONG"
+		}
```

Comment:
🐒 Optional nitpick:
* Can simplify
* 🌴 This change probably applies throughout this PR
* Noting that we don't handle the side or margin type falling outside your boundaries, but we didn't before, either.
```suggestion
	    side := "BUY"
		positionSide := "LONG"
		if s.Side.IsShort() {
			side = "SELL"
			positionSide = "SHORT"
		}
```